### DISCUSSION TITLE
Can anyone provide the documentation for the GATConv code? The code and terminology are slightly different than the Graph Attention Networks.

### DISCUSSION THREAD

**Discussion by arake001** [pyg-team/pytorch_geometric#2930] [Q&A]
Can anyone provide the documentation for the GATConv code? The code and terminology are slightly different than the Graph Attention Networks.

**Comment by rusty1s** [comment]
I updated the [`GATConv`](https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/nn/conv/gat_conv.py) in order to make it more readable and added some more comments/documentation.

Here, `self.lin` refers to the weight matrix **W** in the paper, and `self.lin(x)` refers to **W** * h. The parameters used for computing attention coefficients are given by `self.att_src` and `self.att_dst`. Note that we separate this computation into two steps: We first compute a scalar value individually for both node-level source and target node representations, and sum them up later in the `message` function. This is equivalent to first concatenating source and target node representations, and apply the dot product afterwards. Let me know if there is anything still unclear to you.
